contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--2.8.0--2.9.0.sql

```diff
@@ -350,6 +350,5087 @@ END;
+                                      '\d+\s*(?:\.)+', TIME_MASKSEP_REGEXP, AMPM_REGEXP, '?', TIME_MASKSEP_REGEXP, '$') AND
+             v_culture ~ 'DE[-_]DE|NN[-_]NO|CS[-_]CZ|PL[-_]PL|RO[-_]RO|SK[-_]SK|SL[-_]SI|BG[-_]BG|RU[-_]RU|TR[-_]TR|ET[-_]EE|LV[-_]LV'))
+        THEN
+            RAISE invalid_datetime_format;
+        END IF;
+
+        v_regmatch_groups := regexp_matches(v_srctimestring, CASE v_culture
+                                                                WHEN 'FI' THEN v_defmask3_fi_regexp
+                                                                ELSE v_defmask3_regexp
+                                                             END, 'gi');
+        v_timestring := v_regmatch_groups[1];
+        v_day := '01';
+        v_month := v_regmatch_groups[2];
+        v_year := CASE
+                     WHEN v_culture IN ('TH-TH', 'TH_TH') THEN v_regmatch_groups[3]::SMALLINT - 543
+                     ELSE v_regmatch_groups[3]::SMALLINT
+                  END;
+
+    ELSIF ((v_srctimestring ~* v_defmask5_regexp AND v_culture <> 'FI') OR
+           (v_srctimestring ~* v_defmask5_fi_regexp AND v_culture = 'FI'))
+    THEN
+        IF (v_culture IN ('AR', 'AR-SA', 'AR_SA') OR
+            (v_srctimestring ~ concat('\s*\d{1,2}\.\s*(?:\.|\d+(?!\d)\s*\.)', TIME_MASKSEP_REGEXP, AMPM_REGEXP, '?', TIME_MASKSEP_REGEXP, '\d{1,2}', MASKSEPTWO_REGEXP,
+                                      TIME_MASKSEP_REGEXP, AMPM_REGEXP, '?', TIME_MASKSEP_REGEXP, '\d{1,2}', MASKSEPTWO_REGEXP,
+                                      TIME_MASKSEP_REGEXP, AMPM_REGEXP, '?', TIME_MASKSEP_REGEXP, '\d{3,4}', TIME_MASKSEP_REGEXP, AMPM_REGEXP, '?', TIME_MASKSEP_REGEXP, '$|',
+                                      '\d{1,2}', MASKSEPTWO_REGEXP, TIME_MASKSEP_REGEXP, AMPM_REGEXP, '?', TIME_MASKSEP_REGEXP, '\d{3,4}\s*(?:\.)+|',
+                                      '\d+\s*(?:\.)+', TIME_MASKSEP_REGEXP, AMPM_REGEXP, '?', TIME_MASKSEP_REGEXP, '$') AND
+             v_culture ~ 'DE[-_]DE|NN[-_]NO|CS[-_]CZ|PL[-_]PL|RO[-_]RO|SK[-_]SK|SL[-_]SI|BG[-_]BG|RU[-_]RU|TR[-_]TR|ET[-_]EE|LV[-_]LV'))
+        THEN
+            RAISE invalid_datetime_format;
+        END IF;
+
+        v_regmatch_groups := regexp_matches(v_srctimestring, v_defmask5_regexp, 'gi');
+        v_timestring := concat(v_regmatch_groups[1], v_regmatch_groups[5]);
+        v_year := CASE
+                     WHEN v_culture IN ('TH-TH', 'TH_TH') THEN v_regmatch_groups[4]::SMALLINT - 543
+                     ELSE v_regmatch_groups[4]::SMALLINT
+                  END;
+
+        IF (v_date_format = 'DMY' OR
+            v_culture IN ('LV-LV', 'LV_LV'))
+        THEN
+            v_day := v_regmatch_groups[2];
+            v_month := v_regmatch_groups[3];
+        ELSE
+            v_day := v_regmatch_groups[3];
+            v_month := v_regmatch_groups[2];
+        END IF;
+
+    ELSIF ((v_srctimestring ~* v_defmask7_regexp AND v_culture <> 'FI') OR
+           (v_srctimestring ~* v_defmask7_fi_reg
```

Comment:
Applies to here and at other places as well: Which error code are we expecting here to catch here which `babelfish_drop_deprecated_object` function can't? Can we include handling of this additional error code in `babelfish_drop_deprecated_object` instead or have your own drop function?

Reply:
Updated `babelfish_drop_deprecated_object` to include handling of object does not exists error code.